collector/indices_mappings.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	defaultIndicesMappingsLabels = []string{"index"}
+)
+
+type indicesMappingsMetric struct {
+	Type  prometheus.ValueType
+	Desc  *prometheus.Desc
+	Value func(indexMapping IndexMapping) float64
+}
+
+// IndicesMappings information struct
+type IndicesMappings struct {
+	logger log.Logger
+	client *http.Client
+	url    *url.URL
+
+	up                              prometheus.Gauge
+	totalScrapes, jsonParseFailures prometheus.Counter
+
+	metrics []*indicesMappingsMetric
+}
+
+// NewIndicesMappings defines Indices IndexMappings Prometheus metrics
+func NewIndicesMappings(logger log.Logger, client *http.Client, url *url.URL) *IndicesMappings {
+	subsystem := "indices_mappings_stats"
+
+	return &IndicesMappings{
+		logger: logger,
+		client: client,
+		url:    url,
+
+		up: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: prometheus.BuildFQName(namespace, subsystem, "up"),
+			Help: "Was the last scrape of the ElasticSearch Indices Mappings endpoint successful.",
+		}),
+		totalScrapes: prometheus.NewCounter(prometheus.CounterOpts{
+			Name: prometheus.BuildFQName(namespace, subsystem, "total_scrapes"),
```

Comment:
Counters should end in `_total`.

```suggestion
			Name: prometheus.BuildFQName(namespace, subsystem, "scrapes_total"),
```

Reply:
total_scrapes is consistent with the other places in the exporter (https://github.com/prometheus-community/elasticsearch_exporter/search?q=total_scrapes).  Should I update all those locations as well?  Or fix just this one since it's new and leave the remainder for a breaking change down the road?

Reply:
Let's open a separate PR to fix the naming issues.

Reply:
FYI, it might be useful to run `curl http://exporter:9114/metrics | promtool check metrics`.

Reply:
👍  sounds good I'll make a separate PR based off this one and fix them all together

Reply:
It's better to make a PR fix one thing at a time, it makes it easier to review and isolate changes.